.github/workflows/validate-pr.yaml

```diff
@@ -108,3 +120,74 @@ jobs:
         with:
           number: ${{ github.event.pull_request.number }}
           path: code-coverage-results.md
+
+  release:
+    name: 🐳 Build & publish Docker image
+    runs-on: ubuntu-latest
+    needs:
+      - lint
+      - type-check
+      - test
+    env:
+      image-name: gh-mpyl
+      image-version: pr-${{ github.event.pull_request.number }}
+    outputs:
+      image: ${{ steps.login-ecr-public.outputs.registry }}/vdb-public/${{ env.image-name }}:${{ env.image-version }}
+    steps:
+      - uses: actions/checkout@v4.2.1
+
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v4.0.2
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ECR_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_ECR_SECRET_ACCESS_KEY }}
+          aws-region: us-east-1
+
+      - name: Login to Amazon ECR Public
+        id: login-ecr-public
+        uses: aws-actions/amazon-ecr-login@v2.0.1
+        with:
+          registry-type: public
+
+      - name: Create repository in ECR
+        uses: int128/create-ecr-repository-action@v1.319.0
+        with:
+          repository: ${{ env.image-name }}
+          public: true
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3.2.0
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3.7.1
+
+      - name: Build and push
+        id: docker-build-push
+        uses: docker/build-push-action@v6.9.0
+        with:
+          file: Dockerfile
+          context: .
+          tags: ${{ steps.login-ecr-public.outputs.registry }}/${{ vars.AWS_ECR_PUBLIC_ALIAS }}/${{ env.image-name }}:${{ env.image-version }}
+          platforms: |-
+            linux/amd64
+            linux/arm64
+          provenance: false
+          push: true
+
+  run:
+    name: ▶️ Run action
+    runs-on: ubuntu-latest
+    needs:
+      - release
+    steps:
+      - uses: actions/checkout@v4.2.1
+
+      - name: Point action.yaml to the new version
+        env:
+          IMAGE: docker://${{ needs.release.outputs.image }}
+        run: yq -i '.runs.image = strenv(IMAGE)' action.yaml
+
+      - name: Run docker image
```

Comment:
Wait, how does this work?

Reply:
That makes sense, thanks!

While we're here, I suppose `command group` (and `command`) could be a list/enum/whatever it's called rather than a free string, but it's out of scope and just a nice to have

```
group:
    description: The command group to run
    required: true
  command:
    description: The command to run
    required: false
```

Reply:
It _could_, but the `choice` type is only supported on `workflow_dispatch` (manual) workflows.